pipelines/scripts/private_endpoint/synapse_managed_private_endpoint_manager.py

```diff
@@ -0,0 +1,39 @@
+from pipelines.scripts.private_endpoint.managed_private_endpoint_manager import ManagedPrivateEndpointManager
+from azure.identity import AzureCliCredential, ChainedTokenCredential
+from typing import Dict, List, Any
+import requests
+
+
+class SynapseManagedPrivateEndpointManager(ManagedPrivateEndpointManager):
+    """
+        Class for fetching managed private endpoints from synapse
+    """
+    credential = None
```

Comment:
Why are those Class variables static ? don't you want them to be moved into instance variable ? self.token / self.credential to avoid race conditions in a multithread tests or some parallel execution ?